### PR TITLE
Moved gitalk config to data/config.js

### PR DESCRIPTION
**Pull request by vindu939**

### CODE REVIEW

**Comment by vindu939** [comment]
@calpa https://github.com/calpa/blog/issues/47

**Comment by calpa** [comment]
Deploy preview for *calpa* ready!

Built with commit 70892deb4b6f54f01293c6a5c2ee2730bf700b16

https://deploy-preview-53--calpa.netlify.com

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,8 @@ yarn-error.log
 
 # Local keys
 .env.development
+### Example user template template
```
Comment (calpa) [inline .gitignore]: Can you have a look at this part?

File: data/config.js [hunk 2]
```diff
@@ -72,4 +72,12 @@ module.exports = {
       toPath: '/page/1',
     },
   ],
+  giTalk: {
```
Comment (calpa) [inline data/config.js]: gitalk is ok

File: src/pages/guestbook/index.js [hunk 3]
```diff
@@ -22,19 +24,13 @@ const Gitalk = isBrowser ? require('gitalk') : undefined;
 class Guestbook extends Component {
   constructor(props) {
     super(props);
-    this.data = this.props.data;
+    const { data } = this.props;
```
Comment (calpa) [inline src/pages/guestbook/index.js]: no need to change it

File: src/pages/guestbook/index.js [hunk 4]
```diff
@@ -12,6 +13,7 @@ import ShareBox from '../../components/ShareBox';
 import SEO from '../../components/SEO';
 
 import wrapLayout from '../../api/layout';
+import { giTalk } from '../../../data/config';
```
Comment (calpa) [inline src/pages/guestbook/index.js]: gitalk

File: src/pages/guestbook/index.js [hunk 5]
```diff
@@ -22,19 +24,13 @@ const Gitalk = isBrowser ? require('gitalk') : undefined;
...
     // Gitalk
-    const gitalk = new Gitalk({
-      clientID: '18255f031b5e11edd98a',
-      clientSecret: '2ff6331da9e53f9a91bcc991d38d550c85026714',
-      repo: 'calpa.github.io',
-      owner: 'calpa',
-      admin: ['calpa'],
-      distractionFreeMode: true,
-    });
+    const gitalk = new Gitalk(giTalk);
```
Comment (calpa) [inline src/pages/guestbook/index.js]: gitalk

File: src/templates/blog-post.js [hunk 6]
```diff
@@ -27,7 +27,9 @@ import ShareBox from '../components/ShareBox';
 import { getUrl } from '../api/url';
 import wrapLayout from '../api/layout';
 
-import { url, name, iconUrl } from '../../data/config';
+import {
+  url, name, iconUrl, giTalk,
```
Comment (calpa) [inline src/templates/blog-post.js]: gitalk

File: src/templates/blog-post.js [hunk 7]
```diff
@@ -71,16 +74,10 @@ class BlogPost extends Component {
...
     }
 
-    const gitalk = new Gitalk({
-      clientID: '18255f031b5e11edd98a',
-      clientSecret: '2ff6331da9e53f9a91bcc991d38d550c85026714',
-      repo: 'calpa.github.io',
-      owner: 'calpa',
-      admin: ['calpa'],
-      distractionFreeMode: true,
+    const gitalk = new Gitalk(Object.assign({}, giTalk, {
```
Comment (calpa) [inline src/templates/blog-post.js]: you can use ES6 Object Spread Syntax, and gitalk also

**Comment by vindu939** [comment]
@calpa made the requested changes please review

File: .gitignore [hunk 8]
```diff
@@ -1,11 +1,3 @@
...
-node_modules
-.cache/
-# Build directory
-public/
-.DS_Store
-yarn-error.log
-
-# Local keys
-.env.development
+.idea/
```
Comment (calpa) [inline .gitignore]: Please revert this file.

File: src/pages/guestbook/index.js [hunk 9]
```diff
@@ -27,15 +29,8 @@ class Guestbook extends Component {
...
-    const gitalk = new Gitalk({
-      clientID: '18255f031b5e11edd98a',
-      clientSecret: '2ff6331da9e53f9a91bcc991d38d550c85026714',
-      repo: 'calpa.github.io',
-      owner: 'calpa',
-      admin: ['calpa'],
-      distractionFreeMode: true,
-    });
-    gitalk.render('gitalk-container');
+    const giTalk = new Gitalk(gitalk);
```
Comment (calpa) [inline src/pages/guestbook/index.js]: giTalk -> gitalk

File: src/pages/guestbook/index.js [hunk 10]
```diff
@@ -27,15 +29,8 @@ class Guestbook extends Component {
...
+    giTalk.render('gitalk-container');
```
Comment (calpa) [inline src/pages/guestbook/index.js]: giTalk -> gitalk

File: src/templates/blog-post.js [hunk 11]
```diff
@@ -71,17 +74,12 @@ class BlogPost extends Component {
...
     }
 
-    const gitalk = new Gitalk({
-      clientID: '18255f031b5e11edd98a',
-      clientSecret: '2ff6331da9e53f9a91bcc991d38d550c85026714',
-      repo: 'calpa.github.io',
-      owner: 'calpa',
-      admin: ['calpa'],
-      distractionFreeMode: true,
+    const giTalk = new Gitalk({
```
Comment (calpa) [inline src/templates/blog-post.js]: giTalk -> gitalk

File: src/pages/guestbook/index.js (hunk 9 above)
Comment (vindu939) [reply src/pages/guestbook/index.js]: @calpa there are two `gitalk` variables in the file one has to be renamed.

File: src/pages/guestbook/index.js (hunk 10 above)
Comment (vindu939) [reply src/pages/guestbook/index.js]: @calpa same goes here too

File: src/templates/blog-post.js (hunk 11 above)
Comment (vindu939) [reply src/templates/blog-post.js]: @calpa and same goes here as well

File: src/pages/guestbook/index.js (hunk 9 above)
Comment (calpa) [reply src/pages/guestbook/index.js]: oh, maybe we can use `GitTalkInstance`?

Comment (vindu939) [reply src/pages/guestbook/index.js]: ya I will change to that

**Comment by vindu939** [comment]
@calpa done with the changes you can review once again

**Comment by calpa** [comment]
@vindu939 Good job! The pull request can be merged now. Close #47 

One more thing, @vindu939 would you update the gitalk session in the README.md? 

Submit in the same pull request or another pull request is ok.

**Comment by vindu939** [comment]
Sure I'll update on the same

**Comment by calpa** [comment]
@vindu939 Thank you and your pull request is merged.

**Pull request merged**